### ISSUE TITLE
Add optional argument to avoid overwriting Jacobian in least square

### ISSUE THREAD

**Issue by santisoler** [fatiando/verde#252]
**Description of the desired feature**

The `verde.base.least_squares` solves a linear problem with Tikhonov zero order regularization by rescaling the Jacobian matrix to unit variance columns, so the damping parameter is independent of the units of the misfit. While scaling the Jacobian, a `copy=False` argument is passed to `StandardScaler` so we can save memory by scaling the Jacobian in place. This has the advantage of saving memory, which is a good feature to have for most cases. But on some examples we may want to reuse the computed Jacobian, so we want to avoid to overwrite it.

Would be nice to have an optional `copy` argument on `verde.base.least_squares` that is passed to the `StandardScaler` constructor.

If we decide not  to add this option, I would suggest to add a warning on the docstring explicitly saying that the Jacobian matrix will be overwritten. Also to change the following line:
https://github.com/fatiando/verde/blob/9884a274fc7cc49e213345d21f17d960082eaef0/verde/base/least_squares.py#L48

to:
```python
scaler.fit_transform(jacobian)
```
because `fit_transform` scales the matrix it in place if `copy` is `False`.
This way we can also figure out that the matrix will be overwritten by just reading the code.

**Are you willing to help implement and maintain this feature?** Yes

**Comment by leouieda** [comment]
Ah, nice catch :+1: Maybe add `copy_jacobian=False`? Since this would double the memory requirement, it might be good to add a warning to the docstring. This is also going to be needed if we do anything parallel that shares the Jacobian between workers (we can't have them all trying to scale the same array).
